lib/new_relic/agent/instrumentation/roda/instrumentation.rb

```diff
@@ -0,0 +1,54 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module Roda
+    module Tracer
+      include ::NewRelic::Agent::Instrumentation::ControllerInstrumentation
+
+      def self.included(clazz)
+        clazz.extend(self)
+      end
+
+      def newrelic_middlewares
+        middlewares = [NewRelic::Rack::BrowserMonitoring]
+        if NewRelic::Rack::AgentHooks.needed?
+          middlewares << NewRelic::Rack::AgentHooks
+        end
+        middlewares
+      end
+
+      def build_rack_app_with_tracing
+        unless NewRelic::Agent.config[:disable_roda_auto_middleware]
+          newrelic_middlewares.each do |middleware_class|
+            self.use middleware_class
+          end
+        end
+        yield
+      end
+
+      # Roda makes use of Rack, so we can get params from the request object
+      def rack_request_params
+        begin
+          @_request.params
+        rescue => e
+          NewRelic::Agent.logger.debug('Failed to get params from Rack request.', e)
+          NewRelic::EMPTY_HASH
+        end
+      end
+
+      def _roda_handle_main_route_with_tracing(*args)
+        request_params = rack_request_params
+        filtered_params = ::NewRelic::Agent::ParameterFiltering::apply_filters(request.env, request_params)
+        name = TransactionNamer.transaction_name(request)
+
+        perform_action_with_newrelic_trace(:category => :roda,
```

Comment:
nit: When breaking up the args for a method across multiple lines, I prefer to put the first arg on the line below the method definition.

Reply:
I'd write it like this (align second argument with first), for a third option. I recognize all 3 as being idiomatic enough to represent "how a Rubyist would write it", so all are fine as long as our RuboCop settings permit them.

```ruby
perform_action_with_newrelic_trace(category: :roda,
                                   name: name,
                                   params: filtered_params) do
```

Reply:
Nice suggestions! I've updated this to the Rubocop preference, which is each arg on its own line + the closing paren on its own line